test/fixtures/served/iframe-stub.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html>
+<body onhashchange="postMessageToParent()" style="background-color:red">
+<script>
+
+  /**
+   * Reads JSON format message data from the URL fragment, and then post it to
+   * parent window.
+   */
+  function postMessageToParent() {
+    if (location.hash.length > 1) {
+      const data = JSON.parse(decodeURIComponent(location.hash.substr(1)));
+      window.parent.postMessage(data, '*');
+      console.log('posted message to parent window: ' + JSON.stringify(data));
+    }
+  }
+
+  postMessageToParent();
+
+  // Echo back a message received from parent window with a marker field
+  // 'testStubEcho'.
+  window.addEventListener('message', event => {
+    if (event.source == window.parent && !event.data.testStubEcho) {
+      const data = Object.assign(event.data, {testStubEcho: true});
```

Comment:
I think it's best to rename the type field or whatever we often use. Otherwise, this can interfere with the functioning of an element. Alternative, and might be a better option is to send the wrapper object with `testStubEcho: true` that will contain `payload` with the original message inside.